File: 217.contains-duplicate.java
/*
 * @lc app=leetcode id=217 lang=java
 *
 * [217] Contains Duplicate
 */

// @lc code=start
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> seen = new HashSet<>();

        for (Integer n: nums){
            if (seen.contains(n)){return true;}
            else {
                seen.add(n);
            }
        }

        return false;
    }
}
// @lc code=end

File: 217.contains-duplicate.cpp
/*
 * @lc app=leetcode id=217 lang=cpp
 *
 * [217] Contains Duplicate
 */

// @lc code=start
#include <vector>
#include <unordered_set>
using namespace std;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> seen;
        for (int n : nums) {
            if (seen.find(n) != seen.end()) return true;
            seen.insert(n);
        }
        return false;
    }
};
// @lc code=end

